src/Entity/EdgeEntityViewBuilder.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Entity;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityViewBuilder;
+
+/**
+ * Default view builder class for Apigee Edge entities.
+ */
+class EdgeEntityViewBuilder extends EntityViewBuilder {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getBuildDefaults(EntityInterface $entity, $view_mode): array {
+    $defaults = parent::getBuildDefaults($entity, $view_mode);
+
+    // Simplified path to the entity.
+    $defaults['#entity'] = $entity;
+
+    // Use the generic apigee_entity theme if the entity type specific theme
+    // does not exist. Drupal core does not provide a default template for
+    // entities yet. This solution also allows to change the default, non-entity
+    // type specific theme that is being used if the entity type specific does
+    // not exist.
+    // When 2808481 gets in core, the default theme becomes the "entity". For
+    // BC, we only use that if it configured on the entity type to be used.
+    // @see apigee-entity.html.twig
+    // @see https://www.drupal.org/project/drupal/issues/2808481
+    if (!isset($defaults['#theme']) || $defaults['#theme'] === 'entity') {
+      /** @var \Drupal\apigee_edge\Entity\EdgeEntityTypeInterface $entity_type */
+      $entity_type = $entity->getEntityType();
+      // Add the default #theme key if a template exists for it.
+      if ($entity_type->getDefaultTheme() !== 'apigee_entity' && $this->themeRegistry->getRuntime()->has($entity_type->getDefaultTheme())) {
```

Comment:
This is a more robust solution that does not just allow to override the default theme used by the entity view builder to render this entity, but else helps to ensure BC after 2808481 gets solved. (See more details in the comment above.)

Reply:
While that issue would help here (with theme hook suggestions, which we can still create on our own), we are still left to deal with what we have in core, and as far as I understand #2808481:

- Is not fully fleshed out.
- Likely not going to make it into 8.x, for BC reasons.
- Is meant to add *additional* sane defaults for entities.

Right now, EntityViewBuilder always uses `$entity_type_id` as `#theme`. If we just use those, and implement theme hook suggestions, instead of trying to make one template do everything, we wouldn't need to do any of this.